### PR TITLE
test: filterToParams unit tests

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-commons-js/pull/15#issuecomment-864383058 |
| Dependencies | -- |
| Decisions | Unit tests for `filterToParams` function. |
| Animated GIF | -- |

### CODE REVIEW

File: js/filter.js [hunk 1]
```diff
@@ -93,9 +93,11 @@ export const filterToParams = (
         // adds the multiple filters that will apply the filter string
         // to the multiple default targeting fields, effectively trying
         // to mach any of them (fuzzy string searching)
+        const flatMap = (f, xs) => xs.reduce((acc, x) => acc.concat(f(x)), []);
```
Comment (gcandal) [inline js/filter.js:96]: Move this to an util or something like that

File: types/util.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export declare function flatMap(f: () => unknown, xs: unknown[]): unknown[];
```
Comment (gcandal) [inline types/util.d.ts:1]: This doesn't seem right. It needs to be something like: `<I, O> flatMap(I => O | O[], I[]): O[]`

File: test/base/util.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+global.window = global;
+
+const assert = require("assert");
+const ripeCommons = require("../..");
+
+describe("flatMap()", function() {
+    it("should be able to create a 1 dimension array", () => {
+        const result = ripeCommons.flatMap(i => [i + 1], [1, 2, 3]);
```
Comment (gcandal) [inline test/base/util.js:8]: Test with `i => i` as well

Comment (gcandal) [reply test/base/util.js:8]: And with `i => [i + 1, i]` or something like that

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**